elixir/apps/web/lib/web/live/resources_live/edit.ex

```diff
@@ -1,6 +1,42 @@
 defmodule Web.ResourcesLive.Edit do
   use Web, :live_view
 
+  alias Domain.Gateways
+  alias Domain.Resources
+
+  def enabled_filters(filters) do
+    accumulator = %{all: false, icmp: false, tcp: false, udp: false}
+
+    Enum.reduce(filters, accumulator, fn f, acc ->
+      Map.put(acc, f.protocol, true)
+    end)
+  end
+
+  def ports(filters, type) do
+    filter = Enum.find(filters, &(&1.protocol == type))
+
+    case filter do
+      nil -> ""
+      _ -> Enum.join(filter.ports, ", ")
+    end
+  end
+
+  def mount(%{"id" => id} = _params, _session, socket) do
+    {:ok, resource} =
+      Resources.fetch_resource_by_id(id, socket.assigns.subject, preload: :gateway_groups)
+
+    {:ok, gateway_groups} = Gateways.list_groups(socket.assigns.subject)
+
+    socket =
+      assign(socket,
+        filter_types: enabled_filters(resource.filters),
```

Comment:
I think here naming feels a bit confusing, if the function `enabled_filters` returns anything than `enabled_filters` as the name suggests then the function name should be different, otherwise the assigned variable name should match the function return name.
```suggestion
        enabled_filters: enabled_filters(resource.filters),
```

Reply:
Yeah, I see where you're coming from.  How would you feel about the function and attr being named `filter_states`?
The goal of the function is to return the state for all available filter types, because when each of the filter types gets listed below as checkboxes, we have to know whether it's currently enabled or not.

Reply:
`filter_states` both for var name and function name sound good too.

Reply:
Updated this to `filter_states`